lhs.js

```diff
@@ -15,6 +15,24 @@ function dom(selector) {
     return new DomLhs(selector);
 }
 
+/*
+ * Of all the dom nodes selected by type() or dom(), return only
+ * the fnodes that satisfy all the predicates imposed by calls to
+ * when()
+ */
+function checkPredicates(predicates, fnodes) {
```

Comment:
I'd rather have this named after the thing it returns. In the Fathom codebase, [as is my usual practice](https://www.grinchcentral.com/function-names-to-verb-or-not-to-verb), I've used imperative names, like "checkFact", to imply no return value and only side effects (like the raising of an exception, in that case). I would suggest `satisfyingWhens` or `fnodesSatisfyingWhens` or something.